library/modules/EventManager.cpp

```diff
@@ -199,6 +199,9 @@ static int32_t lastJobId = -1;
 //job completed
 static unordered_map<int32_t, df::job*> prevJobs;
 
+//active units
+static unordered_set<int32_t> activeUnits;
```

Comment:
> Also local static does not work here as it needs to be cleared on map unload and etc.

We've raised the scope of `activeUnits` so that it can be seen and cleared when those units are cleared [on map unload] alongside all the other structures for event manager. So its below declaration has been stricken from the manager function.